src/main/scala/com/fulcrumgenomics/vcf/filtration/ReadEndSomaticVariantFilter.scala

```diff
@@ -123,27 +124,117 @@ class EndRepairArtifactLikelihoodFilter(val distance: Int = 2,
+  extends ReadEndSomaticVariantFilter {
+
+  val Info = VcfInfoHeader("ERP", VcfCount.Fixed(1), VcfFieldType.Float, "P-Value for the End Repair Artifact Filter test.")
+  val Filter = VcfFilterHeader("EndRepairArtifact", "Variant is likely an artifact caused by end repair.")
+
+  override val VcfInfoLines: Iterable[VcfInfoHeader]     = Seq(Info)
+  override val VcfFilterLines: Iterable[VcfFilterHeader] = Seq(Filter)
+
+  /** Applies to all het SNVs. */
+  def appliesTo(gt: Genotype): Boolean = isPointMismatch(gt)
+
+  /** Returns true if the position of the base from the closest read end does not exceed the filter's defined distance. */
+  def isArtifactCongruent(refAllele: Byte, altAllele: Byte, entry : BaseEntry, pileupPosition: Int): Boolean = {
+
+    val positionInRead       = entry.positionInReadInReadOrder
+    val positionFromOtherEnd = Bams.positionFromOtherEndOfTemplate(entry.rec, pileupPosition)
+
+    // Pick the position that's closest to an end, and test for congruency based on that
+    val pos = {
+      if (positionFromOtherEnd.exists(_ < positionInRead)) positionFromOtherEnd.get
+      else                                                 positionInRead
+    }
+
+    pos <= this.distance
+  }
+}
+
+/**
+  * Filter that examines SNVs with the alt allele being A or T to see if they are likely the result
+  * of aberrant A-base addition.
+  *
+  * Occurs when end-repair is performed in a way that chews back single-stranded 3' overhangs
+  * to create a blunt end, but over-digests and ends up creating a recessed 3' end which subsequently
+  * (and incorrectly) gets filled in with one or more As during the A-base addition step. The result
+  * is one or more base substitutions to A at the 3' end of molecules, which after PCR also shows up
+  * as substitutions to T at the 5' end.
+  *
+  * @param distance The distance from the end of the template that defines the region where this
+  *                 artifact may occur
+  * @param pValueThreshold the pvalue at and below which mutations are filtered as being likely
+  *                        caused by the artifact
+  */
+class ATailingArtifactLikelihoodFilter(override val distance: Int = 2, override val pValueThreshold: Option[Double] = None)
+  extends ReadEndSomaticVariantFilter {
+
+  val Info = VcfInfoHeader("ATP", VcfCount.Fixed(1), VcfFieldType.Float, "P-Value for the A-tailing Artifact Filter test.")
+  val Filter = VcfFilterHeader("ATailingArtifact", "Variant is likely an artifact caused by A-tailing.")
+
+  override val VcfInfoLines: Iterable[VcfInfoHeader]     = Seq(Info)
+  override val VcfFilterLines: Iterable[VcfFilterHeader] = Seq(Filter)
+
+  /** Only applies to het SNVs where the alt allele is A or T. */
+  def appliesTo(gt: Genotype): Boolean = {
+    isPointMismatch(gt) &&
+      gt.calls.iterator.filter(_ != gt.alleles.ref).exists(a => a.value.equalsIgnoreCase("A") || a.value.equalsIgnoreCase("T"))
+  }
```

Comment:
Even if `isPointMismatch` were correct, this changes behavior.  The old code required `!gt.isHetNonRef` which requires that the genotype is a) heterozygous and b) one of the alleles in the genotype is the reference allele.  This now only checks that the genotype is heterozygous and one of the called alleles is a non-ref is A or T, but not that other is REF.